hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionProcessor.java

```diff
@@ -60,59 +64,113 @@ public TransactionProcessor(
         this.contractCreation = requireNonNull(contractCreation);
     }
 
+    /**
+     * Process the given transaction, returning the result of running it to completion
+     * and committing to the given updater.
+     *
+     * @param transaction     the transaction to process
+     * @param updater         the world updater to commit to
+     * @param feesOnlyUpdater if base commit fails, a fees-only updater
+     * @param context         the context to use
+     * @param tracer          the tracer to use
+     * @param config          the node configuration
+     * @return the result of running the transaction to completion
+     */
     public HederaEvmTransactionResult processTransaction(
             @NonNull final HederaEvmTransaction transaction,
             @NonNull final HederaWorldUpdater updater,
+            @NonNull final Supplier<HederaWorldUpdater> feesOnlyUpdater,
             @NonNull final HederaEvmContext context,
             @NonNull final HederaTracer tracer,
             @NonNull final Configuration config) {
+        final InvolvedParties parties;
+        final GasCharges gasCharges;
         try {
-            // Compute the sender, relayer, and to address (will throw if invalid)
-            final var parties = computeInvolvedParties(transaction, updater, config);
+            // Compute the sender, relayer, and to address (throws if invalid)
+            parties = computeInvolvedParties(transaction, updater, config);
             if (transaction.isEthereumTransaction()) {
                 parties.sender().incrementNonce();
             }
+            // Charge gas and return intrinsic gas and relayer allowance used (throws on failure)
+            gasCharges = gasCharging.chargeForGas(parties.sender(), parties.relayer(), context, updater, transaction);
+        } catch (final HandleException failure) {
+            return HederaEvmTransactionResult.abortFor(failure.getStatus());
+        }
 
-            // Charge gas and return intrinsic gas and relayer allowance used (will throw on failure)
-            final var gasCharges =
-                    gasCharging.chargeForGas(parties.sender(), parties.relayer(), context, updater, transaction);
+        // Build the initial frame for the transaction
+        final var initialFrame = frameBuilder.buildInitialFrameWith(
+                transaction,
+                updater,
+                context,
+                config,
+                parties.sender().getAddress(),
+                parties.toAddress(),
```

Comment:
This right here indicates that maybe the "to address" is misnamed because when you use this getter it makes it look like you're _converting_ `parties` "to an address" (instead of indicating that you're getting the "to address").

Reply:
Renamed to `receiverAddress`.